superset/datasets/commands/duplicate.py

```diff
@@ -44,60 +43,51 @@
 
 class DuplicateDatasetCommand(CreateMixin, BaseCommand):
     def __init__(self, data: Dict[str, Any]) -> None:
-        self._base_model: SqlaTable = SqlaTable()
+        self._base_model: Optional[SqlaTable] = None
         self._properties = data.copy()
+        self._database: Optional[Database] = None
 
     def run(self) -> Model:
         self.validate()
         try:
-            database_id = self._base_model.database_id
-            table_name = self._properties["table_name"]
-            owners = self._properties["owners"]
-            database = db.session.query(Database).get(database_id)
-            if not database:
-                raise SupersetErrorException(
-                    SupersetError(
-                        message=__("The database was not found."),
-                        error_type=SupersetErrorType.DATABASE_NOT_FOUND_ERROR,
-                        level=ErrorLevel.ERROR,
-                    ),
-                    status=404,
-                )
-            table = SqlaTable(table_name=table_name, owners=owners)
-            table.database = database
-            table.schema = self._base_model.schema
-            table.template_params = self._base_model.template_params
-            table.is_sqllab_view = True
-            table.sql = ParsedQuery(self._base_model.sql).stripped()
-            db.session.add(table)
-            cols = []
-            for config_ in self._base_model.columns:
-                column_name = config_.column_name
-                col = TableColumn(
-                    column_name=column_name,
-                    verbose_name=config_.verbose_name,
-                    expression=config_.expression,
-                    filterable=True,
-                    groupby=True,
-                    is_dttm=config_.is_dttm,
-                    type=config_.type,
-                    description=config_.description,
-                )
-                cols.append(col)
-            table.columns = cols
-            mets = []
-            for config_ in self._base_model.metrics:
-                metric_name = config_.metric_name
-                met = SqlMetric(
-                    metric_name=metric_name,
-                    verbose_name=config_.verbose_name,
-                    expression=config_.expression,
-                    metric_type=config_.metric_type,
-                    description=config_.description,
-                )
-                mets.append(met)
-            table.metrics = mets
-            db.session.commit()
+            if self._base_model and self._database:
```

Comment:
Forced to use a check before so `pre-commit` doesn't complain when assigning the properties to the new `dataset`. The majority of the code inside this block remains untouched.

Reply:
@betodealmeida or @hughhhh what do you think here? The validate method will raise if self._base_model or self._database are falsy. Technically those values couldn't be false on line 53. I'm leaning for readability sake to just skip the pypy warning so that the reader understands that those values couldn't be falsy here.

Reply:
Hey @eschutho I changed it now, based on your comments. Thanks